### ISSUE TITLE
java.net.ConnectException: Connection refused Running maestro

### ISSUE THREAD

**Issue by brandon-gs** [mobile-dev-inc/maestro#1728]
**Describe the bug**
I installed maestro 1 day ago and it works, but now every time I try to run a maestro studio command I get this error:

java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.connect0(Native Method)
	at java.base/sun.nio.ch.Net.connect(Net.java:579)
	at java.base/sun.nio.ch.Net.connect(Net.java:568)
	at java.base/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:593)
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:327)
	at java.base/java.net.Socket.connect(Socket.java:639)
	at java.base/java.net.Socket.connect(Socket.java:588)
	at java.base/java.net.Socket.<init>(Socket.java:512)
	at java.base/java.net.Socket.<init>(Socket.java:292)
	at dadb.adbserver.AdbServer.listDadbs(AdbServer.kt:67)
	at dadb.adbserver.AdbServer.listDadbs$default(AdbServer.kt:60)
	at dadb.Dadb$Companion.list(Dadb.kt:244)
	at dadb.Dadb$Companion.list$default(Dadb.kt:243)
	at maestro.cli.device.DeviceService.listAndroidDevices(DeviceService.kt:140)
	at maestro.cli.device.DeviceService.listDevices(DeviceService.kt:124)
	at maestro.cli.device.DeviceService.listConnectedDevices(DeviceService.kt:114)
	at maestro.cli.device.PickDeviceInteractor.pickDeviceInternal(PickDeviceInteractor.kt:40)
	at maestro.cli.device.PickDeviceInteractor.pickDevice(PickDeviceInteractor.kt:17)
	at maestro.cli.session.MaestroSessionManager.selectDevice(MaestroSessionManager.kt:116)
	at maestro.cli.session.MaestroSessionManager.newSession(MaestroSessionManager.kt:65)
	at maestro.cli.command.StudioCommand.call(StudioCommand.kt:51)
	at maestro.cli.command.StudioCommand.call(StudioCommand.kt:19)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1933)
	at picocli.CommandLine.access$1200(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2332)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2326)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2291)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2159)
	at maestro.cli.DisableAnsiMixin$Companion.executionStrategy(DisableAnsiMixin.kt:22)
	at picocli.CommandLine.execute(CommandLine.java:2058)
	at maestro.cli.AppKt.main(App.kt:117)

**To Reproduce**
Steps to reproduce the behavior, e.g.:

1. Start the simulator
2. Run `maestro test ...` or `maestro studio`

**Expected behavior**
Maestro should run the test or open the studio 

**Screenshots**

**Environment information (please complete the following information):**

- Maestro version [e.g. v1.36.0]
- Platform: Android 14
- Framework: React Native
- Device model and OS version: Android Pixels 5 (emulator)
- Emulator
- Host Mac M2

**Comment by sdfgsdfgd** [comment]
that means your adb server or adbd binaries are having problems - not maestro. 

run `adb devices` and use emulator's ID to open a shell to it like `adb -s "emulator-5554" shell`  and see if you can reach them, get to it's logcat. If you can get there while maestro can't connect, then take a look at the emulator's logcat while trying to connect to see any relevant logs. 

If still no luck, run adb kill-server from terminal , then run adb start-server and see if that solves your problems. 

Or simply restart it, it sounds like your adb executables are stuck or something.

**Comment by vynguyenbao28** [comment]
I was having the same problem, before I installed ADB on the window, not the WSL. So when Run` adb devices` did not accept any virtual machine. After I linked the path from the window to the command of  `sudo ln -s /mnt/c/Users/{YOUR-USERNAME}/AppData/Local/Android/Sdk/platform-tools/adb.exe /usr/bin/adb` . Then when running the adb devices command realized the virtual machine, but the Maestro command was involved in the title. Who has a fix?
